plush.go

```diff
@@ -32,24 +33,25 @@ func BuffaloRenderer(input string, data map[string]interface{}, helpers map[stri
 	if err != nil {
 		return "", err
 	}
-	if helpers != nil {
-		for k, v := range helpers {
-			data[k] = v
-		}
+
+	for k, v := range helpers {
+		data[k] = v
 	}
+
 	return t.Exec(NewContextWith(data))
 }
 
 // Parse an input string and return a Template, and caches the parsed template.
 func Parse(input string) (*Template, error) {
 	moot.Lock()
 	defer moot.Unlock()
-	if t, ok := cache[input]; ok {
+	t, ok := cache[input]
```

Comment:
The mutex and cache checking can be avoided altogether if cache is disabled. Adds some redundant code but you could do at top of method:
```
if !cacheEnabled {
    return NewTemplate(input)
}
```
Then all the other `cacheEnabled` checks can be removed.

Reply:
Yes, you're right. Thanks for it.

Reply:
Updated.